delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/Journal.scala

```diff
@@ -18,9 +19,12 @@ import monix.bio.{IO, Task, UIO}
   */
 private[testkit] class Journal[Id, E <: Event] private (
     events: IORef[Vector[Envelope[E]]],
+    eventsByTag: IORef[Map[String, Vector[Envelope[E]]]],
     semaphore: IOSemaphore,
-    entityType: String
-)(implicit idLens: Lens[E, Id]) {
+    entityType: String,
+    tagger: E => Set[String]
+)(implicit idLens: Lens[E, Id])
+    extends EventLog[Envelope[E]] {
```

Comment:
If we have an dummy/test implementation for Aggregate and KeyValueStore, the Dummy and the Service implementation will get quite similar :)